modules/javafx.graphics/src/main/java/com/sun/javafx/tk/quantum/WindowStage.java

```diff
@@ -414,12 +414,12 @@ static Image findBestImage(java.util.List icons, int width, int height) {
                     //larger than 1/scaleFactor
                     double scaleDivider = Math.ceil(1.0 / scaleFactor);
                     scaleFactor = 1.0 / scaleDivider;
-                    adjw = (int)Math.round((double)iw / scaleDivider);
-                    adjh = (int)Math.round((double)ih / scaleDivider);
+                    adjw = (int)Math.round(iw / scaleDivider);
+                    adjh = (int)Math.round(ih / scaleDivider);
                     scaleMeasure = 1.0 - 1.0 / scaleDivider;
                 }
-                double similarity = ((double)width - (double)adjw) / (double)width +
-                    ((double)height - (double)adjh) / (double)height + //Large padding is bad
+                double similarity = (width - adjw) / (double) width +
+                    (height - adjh) / (double) height + //Large padding is bad
```

Comment:
This change _must_ be reverted. There are cases where doing integer arithmetic on intermediate results is not equivalent to doing double arithmetic.

Consider this:

```
int i = Integer.MAX_VALUE;
int j = -1;
double d1 = (double) i - (double) j;
double d2 = i - j;
```

`d1` will be `2147483647` and `d2` will be `-2147483647`.

I can't say whether it is possible in practice, but this change is not acceptable, and is a good example of the general concern I raised.

Reply:
If the casts in the numerator actually matter, then the cast in the denominator can be removed. The latter are the ones that Eclipse flags for me as unnecessary.

Reply:
I still think that any change here would be a very low value change. If done incorrectly, as it was in the initial attempt, it can introduce bugs. Even if done correctly, I see no point in it.

Reply:
I don't really mind either way. Eclipse's refactoring are safe, so I don't see any risk in doing what it says (removing the denominator casts). For me, the less casts the more clear what the calculation is doing. Like I said before, it's probably a personal thing.

Reply:
@kevinrushforth This has been mostly reverted, and is a lesson for me to not do more than what the warning suggests without checking very carefully. It just looked like a bunch of unnecessary casts, but it is has one edge case with negation overflow of a signed integer that I missed (whether it can happen in practice I did not analyze further).